client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/hooks/useStocktakeLineEdit.ts

```diff
@@ -1,31 +1,40 @@
-import { RecordPatch, ArrayUtils } from '@openmsupply-client/common';
-import { ItemRowFragment } from '@openmsupply-client/system';
 import {
-  StocktakeLineFragment,
-  UpsertStocktakeLinesMutation,
-  useStocktake,
-} from './../../../../api';
+  RecordPatch,
+  ArrayUtils,
+  useTranslation,
+  noOtherVariants,
+} from '@openmsupply-client/common';
+import { ItemRowFragment } from '@openmsupply-client/system';
+import { StocktakeLineFragment, useStocktake } from './../../../../api';
 import { DraftStocktakeLine, DraftLine } from '../utils';
 import { useNextItem } from './useNextItem';
 import { useDraftStocktakeLines } from './useDraftStocktakeLines';
+import { useStocktakeLineErrorContext } from 'packages/inventory/src/Stocktake/context';
 
 interface useStocktakeLineEditController {
   draftLines: DraftStocktakeLine[];
   update: (patch: RecordPatch<StocktakeLineFragment>) => void;
   addLine: () => void;
-  save: (lines: DraftStocktakeLine[]) => Promise<UpsertStocktakeLinesMutation>;
+  save: () => Promise<{ errorMessages?: string[] }>;
   isLoading: boolean;
   nextItem: ItemRowFragment | null;
-  isError: boolean;
 }
 
+export const errorMessage = (error: unknown): string => {
+  // Bugsnag it ?
+  if (error instanceof Error) return error.message;
+  return String(error);
+};
+
 export const useStocktakeLineEdit = (
   item: ItemRowFragment | null
 ): useStocktakeLineEditController => {
+  const t = useTranslation(['inventory']);
```

Comment:
no need for the array when using a single namespace
```suggestion
  const t = useTranslation('inventory');
```

Reply:
https://github.com/openmsupply/open-msupply/pull/1252#discussion_r1121131462